zerver/lib/bugdown/__init__.py

```diff
@@ -353,6 +353,22 @@ def __init__(self, md, bugdown):
         self.bugdown = bugdown
         markdown.treeprocessors.Treeprocessor.__init__(self, md)
 
+    def get_actual_image_url(self, url):
+        # type: (Text) -> Text
+        # Add specific per-site cases to convert image-preview urls to image urls.
+        # See https://github.com/zulip/zulip/issues/4658 for more information
+        parsed_url = urllib.parse.urlparse(url)
+        if (parsed_url.netloc == 'github.com' or parsed_url.netloc.endswith('.github.com')):
+            # https://github.com/zulip/zulip/blob/master/static/images/logo/zulip-icon-128x128.png
+            # https://raw.githubusercontent.com/zulip/zulip/master/static/images/logo/zulip-icon-128x128.png
+            if "/blob/" in url:
+                repopath = url.split('github.com')[1]
+                repopath = repopath.replace('blob/', '')
+                newurl = 'https://raw.githubusercontent.com' + repopath
```

Comment:
might be nicer to use urljoin here, and to return on this line (rather than creating the `newurl` variable)